packages/mobx-react-lite/__tests__/useLocalStore.test.tsx

```diff
@@ -475,25 +478,39 @@ describe("enforcing actions", () => {
         })
         expect(result.error).not.toBeDefined()
     })
+
     it("'always' should work", () => {
         mobx.configure({ enforceActions: "always" })
+        consoleWarnMock = jest.spyOn(console, "warn").mockImplementation(() => {})
+
         const { result } = renderHook(() => {
             const [multiplier, setMultiplier] = React.useState(2)
-            const store = useLocalObservable(() => ({
-                multiplier,
-                count: 10,
-                get multiplied() {
-                    return this.multiplier * this.count
-                },
-                inc() {
-                    this.count += 1
-                }
-            }))
+
+            const [store] = useState(() =>
```

Comment:
I am sorry as I've probably misguided you, but this must use `useLocalObservable`, otherwise the test seems useless.
However I think these `enforceAction` tests are actually wrong. 
Take a look at the expectation: `expect(result.error).not.toBeDefined()`. I guess it was supposed to test that there are no errors/warnings, when calling a method (action)?
The effect  with `store.multiplier = multiplier` seems to be incorrectly copy pasted from other tests and since it was passing (due to incorrect expectation) it was left as is?
Also `"only when 'observed' should work"` and `"'never' should work"` are not tested correctly, because the `store.multipler` isn't used anywhere in the `render` (if it would then `'observed'` would actually warn as well).
What do you think, does it seem like a plausible explanation?

Reply:
Ok, it's clear for me that these `enforceAction` tests should be reconsidered. But why useLocalObservable is required here? If we test `enforceAction` there is no difference between `useLocalObservable` and `useState(() => observable(...))`

Reply:
> If we test enforceAction 

It's not supposed to test `enforceAction`, but `useLocalObservable` compatibility with `enforceAction`. 

> no difference between  useLocalObservable and useState(() => observable(...))

It's an implementation detail. IIRC `useLocalObservable` (or `useLocalStore`) did not use `action` for functions in the past, because we did not have `autoAction`. This was causing troubles, so these test were probably supposed to prove it's no longer an issue.
You're correct that since v6, `useLocalObservable` is defined as an alias for `useState(() => observable(...))`. So you could in principle rewrite all test in this file using `useState(() => observable(...))`, move them to a more appropriate location and eventually introduce a test proving that `useLocalObservable` is an alias for `useState(() => observable(...))`. Your call. One way or another these 3 test should be probably changed into something like:
```javascript
configure({ enforceActions: "..." })
const TestComponent = observer(() => {
  const obs = useLocalObservable(() => ({
    x: 1,
    inc() {
      obs.x += 1
    }
  }))

  useEffect(() => obs.inc()) // or replace with click handler if you want
  
  return obs.x;  // must subscribe
})
render(<TestComponent />)
// TODO expect it doesn't warn/throw
```
It shouldn't warn so no issue with unstable snapshot.

Reply:
One more thing, these test should also finally call `_resetGlobalState()`.

Reply:
@urugator 

I've updated the PR. The summary of my changes:
- I've extracted `enforceAction` tests into a separate file
- Added `_resetGlobaState` after each test of this file
- Added `observer` and `useLocalObservable` as suggested + added console check after each test to make sure that there is actually a warning